osf/management/commands/export_user_account.py

```diff
@@ -0,0 +1,264 @@
+            os.mkdir(current_dir)
+            export_metadata(node, current_dir)
+            # export the associated project (metadata, files, wiki, etc)
+            current_dir = '{}{}/'.format(dir, node.node._id)
+            os.mkdir(current_dir)
+            export_node(node.node, user, current_dir)
+            progress.increment()
+        progress.stop()
+    else:
+        progress.start(nodes_to_export.count(), nodes_type.upper())
+        for node in nodes_to_export:
+            current_dir = '{}{}/'.format(dir, node._id)
+            os.mkdir(current_dir)
+            export_node(node, user, current_dir)
+            progress.increment()
+        progress.stop()
+
+def export_account(user_id, only_private=False, only_admin=False, export_files=True, export_wikis=True):
+    """
+    Exports (as a zip file) all of the projects, registrations, and preprints for which the given user is a contributor.
+
+    The directory structure of the exported file is:
+
+    <user_fullname> (<user_guid>).zip
+        preprints/
+            <preprint_guid>/
+                metadata.json
+                <project_guid>/
+                    metadata.json
+                    files/
+                        osfstorage-archive.zip
+                    wikis/
+                        <wiki_page_name>.md
+
+        projects/
+            <project_guid>/
+                metadata.json
+                files/
+                    osfstorage-archive.zip
+                wikis/
+                    <wiki_page_name>.md
+                components/
+                    <component_guid>/
+                        metadata.json
+                        files/
+                        wikis/
+                    <component_guid>/
+                        metadata.json
+                        files/
+                        wikis/
+                        components/
+                            ...
+        registrations/
+            *same as projects*
+
+    """
+    user = OSFUser.objects.get(guids___id=user_id)
+
+    base_dir = '{}/'.format(user_id)
+    preprints_dir = '{}preprints/'.format(base_dir)
+    projects_dir = '{}projects/'.format(base_dir)
+    registrations_dir = '{}registrations/'.format(base_dir)
+
+    os.mkdir(base_dir)
+    os.mkdir(preprints_dir)
+    os.mkdir(projects_dir)
+    os.mkdir(registrations_dir)
+
+    preprints_to_export = (PreprintService.objects
+        .filter(node___contributors__guids___id=user_id)
+        .select_related('node')
+    )
+
+    preprint_projects_exported = [preprint.node._id for preprint in preprints_to_export]
+    projects_to_export = (user.nodes
+        .filter(is_deleted=False, type='osf.node')
+        .exclude(guids___id__in=preprint_projects_exported)
+        .include('guids')
+        .get_roots()
+    )
+
+    registrations_to_export = (user.nodes
+        .filter(is_deleted=False, type='osf.registration', retraction__null=True, embargo__null=True)
```

Comment:
Yeah, should totally be `isnull`.

Not sure what I was doing excluding embargoes, seems fine to leave them in. I'll remove that and also add the embargo date fields to `REGISTRATION_EXPORT_FIELDS`.